File: CPP/S3_ARRAY/L1_EASY/3_isSorted.c
#include<stdio.h>
int isSorted(int arr[],int n)
{
    if(arr[1]>arr[0])
    {
        for(int i=0;i<n-1;i++)
        {
            if(arr[i+1]<arr[i])return 0;
        }
        return 1;
    }
    for(int i=0;i<n-1;i++)
    {
        if(arr[i+1]>arr[i])return 0;
    }
    return 1;
}
int main()
{
    int n;
    scanf("%d",&n);
    int arr[n];
    for(int i=0;i<n;i++)
    {
        scanf("%d",&arr[i]);
    }
    int r=isSorted(arr,n);
    if(r==0)printf("no");
    else printf("yes");
    return 0;

}

File: python/S3_ARRAY/L1_EASY/3_isSorted.py
class math:
    def __init__(self,arr):
        self.arr=arr
    def isSorted(self):
        n=len(self.arr)
        if(n<=2):
            return 1
        if self.arr[0]>self.arr[1]:
            for i in range(n-1):
                if self.arr[i+1]>self.arr[i]:
                    return 0
            return 1
        for i in range(n-1):
            if self.arr[i+1]<self.arr[i]:
                return 0
        return 1
if __name__=="__main__":
    n=input("enter array size: ")
    arr=list(map(int,input("enter array element :").split()))
    m=math(arr)
    r=m.isSorted()
    if r==0:
        print("no")
    else:
        print("yes")